src/main/java/com/microsoft/sqlserver/jdbc/Util.java

```diff
@@ -970,7 +970,7 @@ static synchronized boolean checkIfNeedNewAccessToken(SQLServerConnection connec
 
     // if driver is for JDBC 42 and jvm version is 8 or higher, then always return as SQLServerPreparedStatement42,
     // otherwise return SQLServerPreparedStatement
-    static boolean use42Wrapper() {
+    public static boolean use42Wrapper() {
```

Comment:
Could you check using System.getProperty("java.specification.version"); and skipping the test if java 7 would solve the issue?